dora/core/server/worker/src/main/java/alluxio/worker/http/HttpLoadOptions.java

```diff
@@ -0,0 +1,233 @@
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.http;
+
+/**
+ * A data model used for describing the HTTP load options.
+ */
+public class HttpLoadOptions {
+
+  static class Builder {
+
+    private HttpLoadOptions mLoadOptions;
+
+    private Builder() {
+      mLoadOptions = new HttpLoadOptions();
+    }
+
+    public void setOpType(OpType opType) {
+      mLoadOptions.setOpType(opType);
+    }
+
+    public void setPartialListing(boolean partialListing) {
+      mLoadOptions.setPartialListing(partialListing);
+    }
+
+    public void setVerify(boolean verify) {
+      mLoadOptions.setVerify(verify);
+    }
+
+    public void setBandWidth(long bandWidth) {
+      mLoadOptions.setBandWidth(bandWidth);
+    }
+
+    public void setProgressFormat(String format) {
+      mLoadOptions.setProgressFormat(format);
+    }
+
+    public void setVerbose(boolean verbose) {
+      mLoadOptions.setVerbose(verbose);
+    }
+
+    public void setLoadMetadataOnly(boolean loadMetadataOnly) {
+      mLoadOptions.setLoadMetadataOnly(loadMetadataOnly);
+    }
+
+    public void setSkipIfExists(boolean skipIfExists) {
+      mLoadOptions.setSkipIfExists(skipIfExists);
+    }
+
+    public OpType getOpType() {
+      return mLoadOptions.getOpType();
+    }
+
+    public boolean isPartialListing() {
+      return mLoadOptions.isPartialListing();
+    }
+
+    public boolean isVerify() {
+      return mLoadOptions.isVerify();
+    }
+
+    public long getBandWidth() {
+      return mLoadOptions.getBandWidth();
+    }
+
+    public String getProgressFormat() {
+      return mLoadOptions.getProgressFormat();
+    }
+
+    public boolean isVerbose() {
+      return mLoadOptions.isVerbose();
+    }
+
+    public boolean isLoadMetadataOnly() {
+      return mLoadOptions.isLoadMetadataOnly();
+    }
+
+    public boolean isSkipIfExists() {
+      return mLoadOptions.isSkipIfExists();
+    }
+
+    public static Builder newBuilder() {
+      return new Builder();
+    }
+
+    public HttpLoadOptions build() {
+      return mLoadOptions;
+    }
+  }
+
+  enum OpType {
+    SUBMIT,
+    STOP,
+    PROGRESS;
+
+    public static OpType of(String type) {
+      switch (type.toLowerCase()) {
+        case "submit":
+          return SUBMIT;
+        case "stop":
+          return STOP;
+        case "progress":
+          return PROGRESS;
+        default:
+          throw new UnsupportedOperationException("Unsupported op type: " + type);
+      }
+    }
+  }
+
+  private OpType mOpType;
+
+  private boolean mPartialListing;
+
+  private boolean mVerify;
+
+  private long mBandWidth;
+
+  private String mProgressFormat;
+
+  private boolean mVerbose;
+
+  private boolean mLoadMetadataOnly;
+
+  private boolean mSkipIfExists;
+
+  private HttpLoadOptions() {
+  }
+
+  private void setOpType(OpType opType) {
+    mOpType = opType;
+  }
```

Comment:
instead of having private setters, you can put all the options as arguments to the constructor.

Reply:
We need a builder pattern, so it would be better to put them into setters, but not constructor since this allows user to set different fields independently. Otherwise, user has to call a constructor setting all the fields at once, this breaks builder pattern.

Reply:
You already have setters on `HttpLoadOptions.Builder`, so you can remove the setters on the `HttpLoadOptions` class.